icicle/backend/cpu/src/hash/cpu_merkle_tree.cpp

```diff
@@ -95,7 +94,11 @@ namespace icicle {
 
           // delete completed_segment_id from the map
           const uint64_t completed_segment_id = completed_segment_idx ^ (completed_layer_idx << 56);
-          m_map_segment_id_2_inputs.erase(completed_segment_id);
+          auto segment = m_map_segment_id_2_inputs.find(completed_segment_id);
+          if (segment != m_map_segment_id_2_inputs.end()) {
+            delete segment->second;
```

Comment:
same comment here. With smart pointers you only need to erase

Reply:
fixed